### ISSUE TITLE
upstream filter: upstream filter couldn't handle body buffer correctly

### ISSUE THREAD

**Issue by wbpcode** [envoyproxy/envoy#37087]
*Title*: *upstream filter: upstream filter couldn't handle body stop iteration correctly*

The upstream filter couldn't handle the body buffer correctly. Because when the connection is ready, it will trigger a `continueDecoding()` anyway.

If this is by design, I think we should document it clearly. Or this is a bug?

**Comment by wbpcode** [comment]
cc @alyssawilk

**Comment by alyssawilk** [comment]
Sorry bit unclear about "couldn't handle buffer body".  If one filter stops decoding part of a request but doesn't StopAllIteration I think it has to be prepared for other filters to ContinueIteration, no?

**Comment by wbpcode** [comment]
> Sorry bit unclear about "couldn't handle buffer body".  If one filter stops decoding part of a request but doesn't StopAllIteration I think it has to be prepared for other filters to ContinueIteration, no?

No, in most cases, a filter return stop at the decodeData because it want to handle the whole body.
